pkg/streamer/file.go

```diff
@@ -194,128 +195,59 @@ func fileSizeUpdated(path string, latestSize int64) (int, error) {
 	}
 }
 
-// relaySubDirUpdated checks whether the relay sub directory updated
-// including file changed, created, removed, etc.
-func relaySubDirUpdated(ctx context.Context, watcherInterval time.Duration, dir string,
+// relayLogUpdatedOrNewCreated checks whether current relay log file is updated or new relay log is created.
+// we check the size of the file first, if the size is the same as the latest file, we assume there is no new write
+// so we need to chenk relay meta file to see if the new relay log is created.
+func relayLogUpdatedOrNewCreated(ctx context.Context, watcherInterval time.Duration, dir string,
 	latestFilePath, latestFile string, latestFileSize int64, updatePathCh chan string, errCh chan error) {
-	var err error
-	defer func() {
-		if err != nil {
-			errCh <- err
-		}
-	}()
-
-	// create polling watcher
-	watcher2 := watcher.NewWatcher()
-
-	// Add before Start
-	// no need to Remove, it will be closed and release when return
-	err = watcher2.Add(dir)
-	if err != nil {
-		err = terror.ErrAddWatchForRelayLogDir.Delegate(err, dir)
-		return
-	}
-
-	err = watcher2.Start(watcherInterval)
-	if err != nil {
-		err = terror.ErrWatcherStart.Delegate(err, dir)
-		return
-	}
-	defer watcher2.Close()
-
-	type watchResult struct {
-		updatePath string
-		err        error
-	}
-
-	result := make(chan watchResult, 1) // buffered chan to ensure not block the sender even return in the halfway
-	newCtx, cancel := context.WithCancel(ctx)
-	defer cancel()
-	go func() {
-		for {
-			select {
-			case <-newCtx.Done():
-				result <- watchResult{
-					updatePath: "",
-					err:        newCtx.Err(),
-				}
+	ticker := time.NewTicker(watcherInterval)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ctx.Done():
```

Comment:
please add a comment for this function to express that the function will be block until send something to the two channel. And send the error here

Reply:
could you please tell me what does this phrase mean? `send the error here`